src/NHibernate/Loader/Loader.cs

```diff
@@ -1424,7 +1424,7 @@ private ColumnNameCache RetreiveColumnNameToIndexCache(DbDataReader rs)
 			if (_columnNameCache == null)
 			{
 				Log.Debug("Building columnName->columnIndex cache");
-				_columnNameCache = new ColumnNameCache(rs.GetSchemaTable().Rows.Count);
+				_columnNameCache = new ColumnNameCache(rs.FieldCount);
```

Comment:
Here GetSchemaTable is completely unnecessary so replace with `FieldCount` which does the same as previous code. This is the main culprit of the issue.